src/buffer/out/textBuffer.cpp

```diff
@@ -445,7 +445,6 @@ size_t TextBuffer::FitTextIntoColumns(const std::wstring_view& chars, til::Coord
     while (dist < len)
     {
         cwd.GraphemeNext(state, chars);
-        dist += state.len;

```

Comment:
We shouldn't mark this grapheme as consumed (`dist`) if we'll early return in the `if` right below.